.github/actions/github-check-secrets-source/action.yml

```diff
@@ -0,0 +1,19 @@
+name: Check Github Actions Source
+description: Fails if actions source is not valid for secrets usage
```

Comment:
A good name should tell the purpose and not the implementation, so if here we are triggering an early failure could be something like `fail-fast-with-dependabot`?

anyway is this really different from just checking `github.actor`?

why do you prefer failing the build in this case? for us always worked well to add a condition at job level to skip it (as described in the README) so you won't even billed for this.

Reply:
the problem with skips is that it can easily lead the PR validation to ignore required checks: we prefer failing all the builds where we need secrets for proper validation